File: examservers/examfront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseurl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  public quizzes(){
    return this.http.get(`${baseurl}/quiz/`);
  }

  public addquiz(quiz: any){
    return this.http.post(`${baseurl}/quiz/`,quiz);
  }
  public deletequiz(quizid: any){
    return this.http.delete(`${baseurl}/quiz/${quizid}`);
  }

  // get single quiz
  public  getQuiz(qid : any){
    return this.http.get(`${baseurl}/quiz/${qid}`)
  }
  // Update quiz
  public updatequiz(quiz:any){
    return this.http.put(`${baseurl}/quiz/`,quiz);
  }
  // get quizzes of single category
  public getquizzesofCategory(cid:any){
    return this.http.get(`${baseurl}/quiz/category/${cid}`);
  }
  // get All Active quizzes
  public getActiveQuizzes(){
    return this.http.get(`${baseurl}/quiz/active`)
  }
  // get Category by active quizzes
  public getCategorybyActive(cid :any){
    return this.http.get(`${baseurl}/quiz/category/active/${cid}`)
  }
}

File: examservers/examservers/src/main/java/com/exam/examservers/service/QuizService.java
package com.exam.examservers.service;

import com.exam.examservers.model.exam.Category;
import com.exam.examservers.model.exam.Quiz;

import java.util.List;
import java.util.Set;

public interface QuizService {

    public Quiz addQuiz(Quiz quiz);
    public Quiz updateQuiz(Quiz quiz);
    public Set<Quiz> getQuizzes();
    public Quiz getQuiz(Long quizid);
    public void deleteQuiz(Long quizid);

    public List<Quiz> getQuizzesofcategory(Category category);

    public List<Quiz> getQuizzesOfActive();
    public List<Quiz> getQuizzesOfcategoryActive(Category category);
}
